src/containers/User/UserArea.tsx

```diff
@@ -104,38 +104,39 @@ const MainMenuBar: React.FC<MainMenuBarProps> = ({
     setDropDownsState((currState) => getNewDropdownsState(basepath, currState))
   }, [pathname])
 
-  const dataViewOptions = dataViews.map(({ code, urlSlug, title, defaultFilter }) => ({
-    key: code,
-    text: title,
-    value: urlSlug + (defaultFilter ? `?${defaultFilter}` : ''),
-  }))
+  const dataViewOptions = constructNestedMenuOptions(dataViews, {
+    filterMethod: () => true,
+    mapMethod: (dataView) => {
+      const { code, urlSlug, title, defaultFilter } = dataView
+      return {
+        key: code,
+        text: title,
+        value: '/data/' + urlSlug + (defaultFilter ? `?${defaultFilter}` : ''),
+      }
+    },
+    getSubmenuMethod: (dataView) => dataView.submenu,
+    onClick: (value) => handleMenuSelect(value, 'DataView'),
+  })
 
-  const templateOptions = templates
-    .filter(({ templateCategory: { uiLocation } }) => uiLocation.includes(UiLocation.List))
-    .sort((t1, t2) => (t1.templateCategory.title > t2.templateCategory.title ? 1 : -1))
-    .map((template) => ({
+  const applicationListOptions = constructNestedMenuOptions(templates, {
+    filterMethod: ({ templateCategory: { uiLocation } }) => uiLocation.includes(UiLocation.List),
+    mapMethod: (template) => ({
```

Comment:
I'm no longer sorting the list by name (it's sorted by code in the query).

Not sure if this is okay, but I'm not sure that alphabetical sorting is the right way to go either. Maybe leave it like this, but if a specific order is requested we can maybe add some kind of custom "priority" field. But not until we know what the requirements would be.